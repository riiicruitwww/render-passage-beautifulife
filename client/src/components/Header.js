import React from 'react';
import './Header.scss';

const Header = () => {
  return (
    <div className="Header">
      <h1 className="Header__title">Santa Toeic</h1>
      <button className="Header__back-btn">돌아가기</button>
    </div>
  )
}

export default Header;