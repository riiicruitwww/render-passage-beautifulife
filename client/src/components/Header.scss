.Header {
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  min-height: 80px;
  color: #ffffff;
  background-color: #151515;

  &__title {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    h1 {
      margin: 0 0 5px 0;
      cursor: pointer;
    }

    span {
      color: #b9b9b9;
    }
  }

  &__back-btn {
    position: absolute;
    left: 0;
    margin-left: 1em;
    padding: 0.5em;
    border: 1px solid;
    border-radius: 20px;
    font-size: 0.9rem;
    color: #b9b9b9;
    cursor: pointer;
  }
}
