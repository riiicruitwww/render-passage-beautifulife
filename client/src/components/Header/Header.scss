.Header {
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  min-height: 80px;
  color: #ffffff;
  background-color: #151515;

  &__title {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    h1 {
      margin: 0 0 5px 0;
      cursor: pointer;
    }

    span {
      color: #b9b9b9;
    }
  }

  button {
    position: absolute;
    padding: 8px;
    border: 1px solid;
    border-radius: 20px;
    font-size: 0.9rem;
    transition: all 0.3s;
  }

  &__back-btn {
    left: 0;
    margin-left: 16px;
    color: #e0e0e0;
    opacity: 0.7;
    cursor: pointer;

    &:hover {
      opacity: 1;
    }
  }

  &__utility-btn {
    right: 0;
    margin-right: 16px;
    color: #474747;
    cursor: not-allowed;

    &.active {
      box-sizing: content-box;
      border-color: #151515;
      color: #ffffff;
      background: linear-gradient(90deg, #ef5e80 0%, #519fe5);
      opacity: 0.9;
      cursor: pointer;

      &:hover {
        opacity: 1;
      }
    }
  }
}
