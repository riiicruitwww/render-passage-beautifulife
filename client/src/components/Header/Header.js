import React from 'react';
import './Header.scss';

const Header = ({
  isAnswerComplete,
  isCheckComplete,
  packType,
  onCheckClick,
  onNextClick,
  onHomeClick
}) => {
  const handleCheckClick = () => {
    if (isAnswerComplete) {
      onCheckClick();
    }
  };

  const handleLogoClick = () => {
    onHomeClick();
  };

  const handleNextClick = () => {
    onNextClick();
  };

  return (
    <div className="Header">
      <div className="Header__title">
        <h1 onClick={handleLogoClick}>Santa Toeic</h1>
        <span>{`(${packType || '...Loading'})`}</span>
      </div>
      <button className="Header__back-btn">돌아가기</button>
      {isCheckComplete ? (
        <button
          className="Header__utility-btn active"
          onClick={handleNextClick}
        >
          다음문제
        </button>
      ) : (
        <button
          className={`Header__utility-btn ${isAnswerComplete ? 'active' : ''}`}
          onClick={handleCheckClick}
          disabled={!isAnswerComplete}
        >
          채점하기
        </button>
      )}
    </div>
  );
};

export default Header;
