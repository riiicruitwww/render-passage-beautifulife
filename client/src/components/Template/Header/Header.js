import React from 'react';
import './Header.scss';

const Header = ({
  isAnswerComplete,
  isCheckComplete,
  packType,
  onCheckClick,
  onNextClick,
  onHomeClick
}) => {
  const handleCheckClick = () => {
    if (isAnswerComplete) {
      onCheckClick();
    }
  };

  return (
    <div className="Header">
      <div className="Header__title">
        <h1 onClick={onHomeClick}>Santa Toeic</h1>
        <span>{`(${packType || '...Loading'})`}</span>
      </div>
      {isCheckComplete ? (
        <button className="Header__utility-btn active" onClick={onNextClick}>
          다음문제
        </button>
      ) : (
        <button
          className={`Header__utility-btn ${isAnswerComplete ? 'active' : ''}`}
          onClick={handleCheckClick}
          disabled={!isAnswerComplete}
        >
          채점하기
        </button>
      )}
    </div>
  );
};

export default Header;
