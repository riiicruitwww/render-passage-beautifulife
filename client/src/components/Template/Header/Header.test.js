import React from 'react';
import { shallow, mount } from 'enzyme';
import Header from './Header';

describe('Header', () => {
  describe('about render', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<Header />);
    });

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('utility-btn active when isAnserComplete is true', () => {
      wrapper.setProps({ isAnswerComplete: true });
      expect(wrapper.find('.Header__utility-btn.active').length).toBe(1);
    });

    it('utility-btn active when isCheckComplete is true', () => {
      wrapper.setProps({ isCheckComplete: true });
      expect(wrapper.find('.Header__utility-btn.active').length).toBe(1);
    });
  });

  describe('about props', () => {
    const props = { testProps: 'test' };
    const wrapper = mount(<Header {...props} />);

    it('allows to set props', () => {
      expect(wrapper.props()).toEqual(props);
    });
  });

  describe('about event handlers', () => {
    const props = {
      onCheckClick: jest.fn(),
      onHomeClick: jest.fn(),
      onNextClick: jest.fn()
    };
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<Header {...props} />);
    });

    it('handleLogoClick works when title clicked', () => {
      wrapper.find('.Header__title > h1').simulate('click');
      expect(props.onHomeClick).toHaveBeenCalledTimes(1);
    });

    it('handleCheckClick works when title clicked', () => {
      wrapper.setProps({ isAnswerComplete: true });
      wrapper.find('.Header__utility-btn.active').simulate('click');
      expect(props.onCheckClick).toHaveBeenCalledTimes(1);
    });

    it('handleNextClick works when title clicked', () => {
      wrapper.setProps({ isCheckComplete: true });
      wrapper.find('.Header__utility-btn.active').simulate('click');
      expect(props.onNextClick).toHaveBeenCalledTimes(1);
    });
  });
});
