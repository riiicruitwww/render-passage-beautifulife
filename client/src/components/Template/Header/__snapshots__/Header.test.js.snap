// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Header__title"
        >
          <h1>
            Santa Toeic
          </h1>
          <span>
            (...Loading)
          </span>
        </div>,
        <button
          className="Header__utility-btn "
          disabled={true}
          onClick={[Function]}
        >
          채점하기
        </button>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Santa Toeic
            </h1>,
            <span>
              (...Loading)
            </span>,
          ],
          "className": "Header__title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Santa Toeic",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Santa Toeic",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "(...Loading)",
            },
            "ref": null,
            "rendered": "(...Loading)",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "채점하기",
          "className": "Header__utility-btn ",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "채점하기",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Header__title"
          >
            <h1>
              Santa Toeic
            </h1>
            <span>
              (...Loading)
            </span>
          </div>,
          <button
            className="Header__utility-btn "
            disabled={true}
            onClick={[Function]}
          >
            채점하기
          </button>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Santa Toeic
              </h1>,
              <span>
                (...Loading)
              </span>,
            ],
            "className": "Header__title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Santa Toeic",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Santa Toeic",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "(...Loading)",
              },
              "ref": null,
              "rendered": "(...Loading)",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "채점하기",
            "className": "Header__utility-btn ",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "채점하기",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
